packages/react/src/hooks/public/use-remote-hosts.ts

```diff
@@ -0,0 +1,49 @@
+import { PublicClient } from '@abstract-money/core'
+import { UseQueryOptions, useQuery } from '@tanstack/react-query'
+import React from 'react'
+import { useConfig } from '../../contexts'
+
+type QueryFnData = Awaited<ReturnType<PublicClient['getRemoteHosts']>>
+
+type QueryError = unknown
+type QueryData = QueryFnData
+type QueryKey = readonly [
+  'remoteHosts',
+  PublicClient | undefined,
+  string | undefined,
+]
+
+type QueryOptions = Omit<
+  UseQueryOptions<QueryFnData, QueryError, QueryData, QueryKey>,
+  'queryFn'
```

Comment:
Since we don't omit the `queryKey` from the options in our queries doesn't this mean it could be "accidentially" (or purposefully) overridden? Is this something we'd like to enable?

Esp sinc the syntax has been changed to glob all the options together instead of having the query key, query function as first args to `useQuery`

Reply:
left to be purposefully overriden, don't see it as a blocker rn.